File: go/src/leetcode/editor/cn/0136_single_number.go
package main

func main() {
	println("136、single_number")
}

// leetcode submit region begin(Prohibit modification and deletion)
func singleNumber(nums []int) int {
	number := 0
	for _, num := range nums {
		number ^= num
	}
	return number
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0136_single_number.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("136、single_number");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        for i in 0..nums.len() {
            res ^= nums[i];
        }
        res
    }
}
//leetcode submit region end(Prohibit modification and deletion)
